lib/spack/spack/util/archive.py

```diff
@@ -238,3 +241,42 @@ def reproducible_tarfile_from_prefix(
                 add_file(tar, file_info, entry.path)
 
         dir_stack.extend(reversed(new_dirs))  # we pop, so reverse to stay alphabetical
+
+
+def _git_prefix(archive_path):
+    # This is an annoying method, but since we always have a prefix and can't gaurantee what
+    # it is we need this.
+    tar = which("tar", required=True)
```

Comment:
It would be nice if we could avoid going out to the filesystem a second time to determine the path for `tar` -- is it too ugly to have it as a parameter to this method since the only caller is computing the same thing anyway?